src/interrupt.rs

```diff
@@ -3,9 +3,29 @@
 //! For the most part, [crate::interrupt::free] is what you want:
 //!
 //! ```
-//! atmega32u4::interrupt::free(|cs| {
+//! avr_device::interrupt::free(|cs| {
 //!     // Interrupts are disabled here
-//! })
+//! });
+//! ```
+//!
+//! To access shared state, Mutex can be used:
+//!
+//! ```
+//! use avr_device::interrupt::Mutex;
+//! use core::cell::Cell;
+//!
+//! // Use Cell, if the wrapped type is Copy.
+//! // Use RefCell, if the wrapped type is not Copy or if you need a mut reference to it for other reasons.
```

Comment:
Actually, you will also need the RefCell to get an immutable reference...

```suggestion
//! // Use RefCell, if the wrapped type is not Copy or if you need a reference to it for other reasons.
```

Reply:
I think we don't need a (Ref)Cell at all, if only an immutable ref is needed. Mutex::borrow() returns an immutable ref.
In fact, we wouldn't even need a Mutex for immutable-only global access.
Or am I missing something?

Reply:
Right, but at some point you'll want to fill the Mutex so you need the mutable access at least once.  I don't think there is way around a RefCell then, even if you only grab immutable references later...

Reply:
Ok. It could be static initialization. But this is theoretical anyway. Almost all uses would need Cell or RefCell mutable. Therefore, I removed the word.